src/transformers/generation/utils.py

```diff
@@ -2397,17 +2398,14 @@ def greedy_search(
                 # did all peers finish? the reduced sum will be 0.0 then
                 if this_peer_finished_flag.item() == 0.0:
                     break
-
-            # prepare model inputs
-            model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
+            
+            self.config.output_attentions = output_attentions
+            self.config.output_hidden_states = output_hidden_states
```

Comment:
Is this one also needed?

Reply:
anywhere we hard code `output_attentions=output_attentions,output_hidden_states=output_hidden_states` IMO